cat/spirv.cat

```diff
@@ -0,0 +1,127 @@
+SPIRV
+
+(* Terms from spirv.als *)
+// stor = [] "set to relation, (s,s) for all s in E, lift a set to a relation (the [_] operator in Herd)"
+// rai-r = r | r^-1
+// sthd = int
+// immpo = po - po;po
+// rc[r] = r?
+// ssg, swg, sqf = same scope id @ subgroup, workgroup, queuefamily
+// inscope = sr
+// sloc = loc
+// sref = vloc
+// scbarinst = syncbar "relates pairs of control barriers, same control barrier dynamic instance"
+// imm-r = r - (r;r+)
+// twoplus-r = r;r+
+
+//???????????
+// chains = new relation
+// ssw = new relation for system-synchronizes-with?
+
+(*******************)
+(*  Auxiliaries *)
+(*******************)
+let mutordatom = [ATOM]; ((loc & vloc & sr) \ id); [ATOM] // mutually ordered atomics (same loc, inscope)
+
+let sctosem = SC0 * SEMSC0 | SC1 * SEMSC1
+let semtosc = SEMSC0 * SC0 | SEMSC1 * SC1
+
+let avvisSem = ((SC0 | SC1) * ((AV | VIS) & DV)) | (sctosem; [SEMAV | SEMVIS])
```

Comment:
I don't think `(sctosem; [SEMAV | SEMVIS])` is equivalent to `(SC0->((SEMAV+SEMVIS)&SEMSC0)) + (SC1->((SEMAV+SEMVIS)&SEMSC1))`. Please double check this

Reply:
Updated to `(SC0 * (SEMAV | SEMVIS) & SEMSC0)| (SC1 * (SEMAV | SEMVIS) & SEMSC1)`